tests/utils/test_process_util.py

```diff
@@ -93,7 +102,74 @@ def test_format_stdout_stderr05(self):
         actual = format_stdout_stderr('', '', 1000)
         self.assertEqual(expected, actual)
 
+    def test_process_stdout_stderr(self):
+        """
+        If the command has no timeout, the process need not be the leader of its own process group.
+        """
+        stdout = "The quick brown fox jumps over the lazy dog.\n"
+        stderr = "The five boxing wizards jump quickly.\n"
+
+        expected = "[stdout]\n{0}\n\n[stderr]\n{1}".format(stdout, stderr)
+
+        cmd = process_cmd_template.format(process_target, stdout, stderr)
+        process = subprocess.Popen(cmd,
+                                   shell=True,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   env=os.environ)
+
+        actual = capture_from_process(process, cmd)
+        self.assertEqual(expected, actual)
+
+    def test_process_timeout(self):
+        cmd = "{0} -t 20".format(process_target)
+        process = subprocess.Popen(cmd,
+                                   shell=True,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   env=os.environ,
+                                   preexec_fn=os.setsid)
+
+        if sys.version_info < (2, 7):
+            self.assertRaises(ExtensionError, capture_from_process_raw, process, cmd, 10)
+        else:
+            with self.assertRaises(ExtensionError) as ee:
+                capture_from_process_raw(process, cmd, 10)
+
+            body = str(ee.exception)
+            if sys.version_info >= (3, 2):
```

Comment:
I would just add this to test helpers so you can avoid the whole assertRegex vs assertRegexpMatches.  Save the reading of an if statement.